homeassistant/components/habitica/__init__.py

```diff
@@ -91,41 +34,9 @@ def has_all_unique_users_names(value):
 )
 
 
-async def async_setup(hass, config):
+async def async_setup(hass: HomeAssistant, config: dict):
     """Set up the Habitica service."""
 
```

Comment:
Can we import the config yaml to the config flow, to make the transition easier for users? Define a step `async_step_import` in the config flow and start the flow with source `SOURCE_IMPORT` here.

https://github.com/home-assistant/core/blob/111c2006c8cf37a0d26995c0048c26780f731017/homeassistant/components/hue/__init__.py#L89-L95